tools/tpm2_send.c

```diff
@@ -70,6 +70,10 @@ static int read_command_from_file(FILE *f, tpm2_command_header **c,
     LOG_INFO("command code: 0x%08x", tpm2_command_header_get_code(command));
 
     ret = fread(command->data, data_size, 1, f);
+    if (feof(f)) {
+        LOG_ERR("Faild to read command body: EOF reached");
+        return -1;
+    }
     if (ret != 1 && ferror(f)) {
```

Comment:
Hi, @AndreasFuchsTPM.  Looks like you should also `free(command)`.  Maybe you can just simplify to `if (ret != 1) {` and then drop the strerror in the logging?  You would lose the specifics though, so maybe that isn't desired.

Reply:
@AndreasFuchsTPM I would agree to @jandryuk  and could merge the PR if the free is added.

Reply:
Done.